File: Tests/ButterflyNet.Tests/HtmlEntityTests.cs
using ButterflyNet.Parser.Modules;
using NUnit.Framework;

namespace ButterflyNet.Parser.Tests {
	[TestFixture]
	public class HtmlEntityTests : WikiToHtmlTest {

		[SetUp]
		public override void SetUp() {
			base.SetUp();
			Parser.ModuleFactory = new ActivatorFactory<IButterflyModule>(new NamedTypeRegistry<IButterflyModule>().RegisterType<HtmlEntityModule>("entity"));
		}

		[Test]
		public void Should_display_named_html_entity() {
			Assert.That(Parser.ParseAndReturn("[:entity|value=hellip]"), Is.EqualTo("<p>&hellip;</p>"));
		}

		[Test]
		public void Should_display_numbered_html_entity() {
			Assert.That(Parser.ParseAndReturn("[:entity|value=#8567]"), Is.EqualTo("<p>&#8567;</p>"));
		}

		[Test]
		[ExpectedException(typeof(ModuleException), ExpectedMessage = "The \"value\" property must be set to a valid HTML entity")]
		public void Should_require_value_property() {
			Parser.Parse("[:entity]");
		}

		[Test]
		[ExpectedException(typeof(ModuleException), ExpectedMessage = "The \"value\" property must be set to a valid HTML entity")]
		public void Should_not_allow_empty_entity_value() {
			Parser.Parse("[:entity|value=]");
		}

		[Test]
		[ExpectedException(typeof(ModuleException), ExpectedMessage = "\"foo!\" is not a valid HTML entity")]
		public void Should_validate_entity_value() {
			Parser.Parse("[:entity|value=foo!]");
		}
	}
}

File: src/js/tests/HtmlEntityTests.js
var htmlEntityTests = function() {
	var parser;
	
	return {
		setup: function() {
			parser = createParser();
		},
		
		tearDown: function() {
			parser = null;
		},
		
		test: function HTML_entity_tests() {
			return [
				function Should_display_named_html_entity() {
					Assert.that(trimLf(parser.parseAndReturn("[:entity|value=hellip]")), Is.equalTo("<p>&hellip;</p>"));
				},
				
				function Should_display_numbered_html_entity() {
					Assert.that(trimLf(parser.parseAndReturn("[:entity|value=#8567]")), Is.equalTo("<p>&#8567;</p>"));
				},
				
				function Should_require_value_property() {
					Assert.willThrow(new Butterfly.ModuleException("The \"value\" property must be set to a valid HTML entity"));
					parser.parse("[:entity]");
				},
				
				function Should_not_allow_empty_entity_value() {
					Assert.willThrow(new Butterfly.ModuleException("The \"value\" property must be set to a valid HTML entity"));
					parser.parse("[:entity|value=]");
				},
				
				function Should_validate_entity_value() {
					Assert.willThrow(new Butterfly.ModuleException("\"<script>malicious code</script>\" is not a valid HTML entity"));
					parser.parse("[:entity|value=<script>malicious code</script>]");
				}
			];
		}
	};
}();

Jarvis.run(htmlEntityTests);